src/Mobs/Path.h

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#ifdef COMPILING_PATHFIND_DEBUGGER
+/* Note: the COMPILING_PATHFIND_DEBUGGER flag is used by Native/WiseOldMan95 to debug this class outside of MCServer.
+This preprocessor flag is never set when compiling MCServer. */
+#include "PathFinderIrrlicht_Head.h"
+#endif
+
+#include <vector>
+#include <queue>
+#include <unordered_map>
+using namespace std;
+
+/* MCServer forward declarations */
+#ifndef COMPILING_PATHFIND_DEBUGGER
+
+// fwd: Vector3.h
+template <typename T> class Vector3;
+typedef Vector3<double> Vector3d;
+
+// fwd: cChunkMap.h
+typedef cItemCallback<cChunk> cChunkCallback;
+#endif
+
+/* Various little structs and classes */
+enum ePathFinderStatus {CALCULATING,  PATH_FOUND,  PATH_NOT_FOUND};
```

Comment:
Since this is new code you might want to use an enum class to avoid namespace pollution.